src/components/ChangePasswordModal.tsx

```diff
@@ -0,0 +1,124 @@
+import PopUp from './PopUp'
+import { Button, ModalProps, Spinner } from 'react-bootstrap'
+import { useForm } from 'react-hook-form'
+import Input from './Input'
+import { useEffect } from 'react'
+
+interface IChangePasswordModal extends ModalProps {}
+
+export interface IChangePasswordModalForm {
```

Comment:
We kinda sorta have a convention going where the form state types are named `IYaddaYaddaFormState`. Would be good to do that here for consistency.
```suggestion
export interface IChangePasswordModalFormState {
```